echopype/mask/api.py

```diff
@@ -493,23 +495,67 @@ def frequency_differencing(
         chanA = chanAB[0]
         chanB = chanAB[1]
 
-    # get the left-hand side of condition
-    lhs = source_Sv["Sv"].sel(channel=chanA) - source_Sv["Sv"].sel(channel=chanB)
+        freqA_pos = np.argwhere(source_Sv.channel.values == chanAB[0]).flatten()[0]
+        freqB_pos = np.argwhere(source_Sv.channel.values == chanAB[1]).flatten()[0]
 
-    # create mask using operator lookup table
-    da = xr.where(str2ops[operator](lhs, diff), True, False)
+    xr_dataarray_attrs = {
+        "mask_type": "frequency differencing",
+        "history": f"{datetime.datetime.utcnow()} +00:00. "
+        "Mask created by mask.frequency_differencing. "
+        f"Operation: Sv['{chanA}'] - Sv['{chanB}'] {operator} {diff}",
+    }
 
-    # assign a name to DataArray
-    da.name = "mask"
+    # If Sv data is not dask array
+    if not isinstance(source_Sv["Sv"].variable._data, dask.array.Array):
+        # get the left-hand side of condition
+        lhs = source_Sv["Sv"].sel(channel=chanA) - source_Sv["Sv"].sel(channel=chanB)
 
-    # assign provenance attributes
-    mask_attrs = {"mask_type": "frequency differencing"}
-    history_attr = (
-        f"{datetime.datetime.utcnow()} +00:00. "
-        "Mask created by mask.frequency_differencing. "
-        f"Operation: Sv['{chanA}'] - Sv['{chanB}'] {operator} {diff}"
-    )
+        # create mask using operator lookup table
+        da = xr.where(str2ops[operator](lhs, diff), True, False)
+
+        # assign a name to DataArray
+        da.name = "mask"
 
-    da = da.assign_attrs({**mask_attrs, **{"history": history_attr}})
+        # assign provenance attributes
+        da = da.assign_attrs(xr_dataarray_attrs)
 
-    return da
+        return da
+    # If Sv data is dask array
+    else:
+        num_column_blocks = int(np.ceil(source_Sv["Sv"].shape[2] / source_Sv["Sv"].chunks[2][0]))
```

Comment:
This grabbing of shape is very particular and assumes that the right dimension is located at `2`. What dimension name are you trying to grab here? `range_sample`?